### PR TITLE
feat: Remove deprecation `DEPPS1`: Native MongoDB syntax in aggregation pipeline

### LINKED ISSUE #8268: Remove deprecation `DEPPS1`: Native MongoDB syntax in aggregation pipeline
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

Deprecation to be removed in Parse Sever 6:
- `DEPPS1`: Native MongoDB syntax in aggregation pipeline 

### Feature / Enhancement Description

Remove deprecation, see [list](https://github.com/parse-community/parse-server/blob/alpha/DEPRECATIONS.md).

### Example Use Case
n/a

### Alternatives / Workarounds
n/a

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Parse Server previously used a custom aggregate syntax.

Closes: #8268

### Approach

Remove old syntax

### TODOs before merging

- [x] Add tests

### CODE REVIEW

**Comment by dblythy** [comment]
Depends on a JS SDK release after the merge of https://github.com/parse-community/Parse-SDK-JS/pull/1642 in order for tests to pass

File: spec/AggregateRouter.spec.js [hunk 1]
```diff
@@ -37,55 +34,53 @@ describe('AggregateRouter', () => {
...
     ];
-    try {
-      AggregateRouter.getPipeline(body);
-    } catch (e) {
-      expect(e.message).toBe('Pipeline stages should only have one key found group, match');
-    }
+    expect(() => AggregateRouter.getPipeline(body)).toThrow(
+      new Parse.Error(
+        Parse.Error.INVALID_QUERY,
+        'Pipeline stages should only have one key. Found $group, $match'
```
Comment (mtrezza) [inline spec/AggregateRouter.spec.js]:
```suggestion
        'Pipeline stages should only have one key but found $group, $match.'
```

File: spec/AggregateRouter.spec.js [hunk 2]
```diff
@@ -37,55 +34,53 @@ describe('AggregateRouter', () => {
...
     };
-    try {
-      AggregateRouter.getPipeline(body);
-    } catch (e) {
-      expect(e.message).toBe('Pipeline stages should only have one key found group, match');
-    }
+    expect(() => AggregateRouter.getPipeline(body)).toThrow(
+      new Parse.Error(
+        Parse.Error.INVALID_QUERY,
+        'Pipeline stages should only have one key. Found $group, $match'
```
Comment (mtrezza) [inline spec/AggregateRouter.spec.js]:
```suggestion
        'Pipeline stages should only have one key but found $group, $match.'
```

File: spec/AggregateRouter.spec.js [hunk 3]
```diff
@@ -161,4 +156,16 @@ describe('AggregateRouter', () => {
     const result = AggregateRouter.getPipeline(body);
     expect(result).toEqual(expected);
   });
+
+  it('should throw with invalid stage', () => {
+    expect(() => AggregateRouter.getPipeline([{ foo: 'bar' }])).toThrow(
+      new Parse.Error(Parse.Error.INVALID_QUERY, `Invalid aggregate stage: foo.`)
```
Comment (mtrezza) [inline spec/AggregateRouter.spec.js]:
```suggestion
      new Parse.Error(Parse.Error.INVALID_QUERY, `Invalid aggregate stage 'foo'.`)
```

File: spec/AggregateRouter.spec.js [hunk 4]
```diff
@@ -161,4 +156,16 @@ describe('AggregateRouter', () => {
...
+    );
+  });
+
+  it('should throw with invalid group', () => {
+    expect(() => AggregateRouter.getPipeline([{ $group: { objectId: 'bar' } }])).toThrow(
+      new Parse.Error(Parse.Error.INVALID_QUERY, `Cannot use objectId in aggregation stage $group.`)
```
Comment (mtrezza) [inline spec/AggregateRouter.spec.js]:
```suggestion
      new Parse.Error(Parse.Error.INVALID_QUERY, `Cannot use 'objectId' in aggregation stage $group.`)
```

File: src/Routers/AggregateRouter.js [hunk 5]
```diff
@@ -83,22 +82,30 @@ export class AggregateRouter extends ClassesRouter {
...
       if (Object.prototype.hasOwnProperty.call(stage[stageName], 'objectId')) {
-        Deprecator.logRuntimeDeprecation({
-          usage: 'The use of objectId in aggregation stage $group',
-          solution: 'Use _id instead.',
-        });
-        stage[stageName]._id = stage[stageName].objectId;
-        delete stage[stageName].objectId;
+        throw new Parse.Error(
+          Parse.Error.INVALID_QUERY,
+          `Cannot use objectId in aggregation stage $group.`
```
Comment (mtrezza) [inline src/Routers/AggregateRouter.js]:
```suggestion
          `Cannot use 'objectId' in aggregation stage $group.`
```

File: src/Routers/AggregateRouter.js [hunk 6]
```diff
@@ -83,22 +82,30 @@ export class AggregateRouter extends ClassesRouter {
 
     return pipeline.map(stage => {
       const keys = Object.keys(stage);
-      if (keys.length != 1) {
-        throw new Error(`Pipeline stages should only have one key found ${keys.join(', ')}`);
+      if (keys.length !== 1) {
+        throw new Parse.Error(
+          Parse.Error.INVALID_QUERY,
+          `Pipeline stages should only have one key. Found ${keys.join(', ')}`
```
Comment (mtrezza) [inline src/Routers/AggregateRouter.js]:
```suggestion
          `Pipeline stages should only have one key but found ${keys.join(', ')}`
```

**Review by mtrezza — CHANGES_REQUESTED** [review]
Just some log rephrasing

**Comment by mtrezza** [comment]
Seems there's still some adjustment necessary after the commits, could you take a look?

File: src/Routers/AggregateRouter.js [hunk 7]
```diff
@@ -83,22 +82,30 @@ export class AggregateRouter extends ClassesRouter {
 
     return pipeline.map(stage => {
       const keys = Object.keys(stage);
-      if (keys.length != 1) {
-        throw new Error(`Pipeline stages should only have one key found ${keys.join(', ')}`);
+      if (keys.length !== 1) {
+        throw new Parse.Error(
+          Parse.Error.INVALID_QUERY,
+          `Pipeline stages should only have one key but found ${keys.join(', ')}`
```
Comment (mtrezza) [inline src/Routers/AggregateRouter.js]:
```suggestion
          `Pipeline stages should only have one key but found ${keys.join(', ')}.`
```

File: spec/AggregateRouter.spec.js (hunk 3 above)
Comment (mtrezza) [inline spec/AggregateRouter.spec.js]:
```suggestion
      new Parse.Error(Parse.Error.INVALID_QUERY, `Invalid aggregate stage 'foo'.`)
```

File: spec/AggregateRouter.spec.js [hunk 8]
```diff
@@ -37,55 +34,53 @@ describe('AggregateRouter', () => {
...
     };
-    try {
-      AggregateRouter.getPipeline(body);
-    } catch (e) {
-      expect(e.message).toBe('Pipeline stages should only have one key found group, match');
-    }
+    expect(() => AggregateRouter.getPipeline(body)).toThrow(
+      new Parse.Error(
+        Parse.Error.INVALID_QUERY,
+        'Pipeline stages should only have one key but found $group, $match'
```
Comment (mtrezza) [inline spec/AggregateRouter.spec.js]:
```suggestion
        'Pipeline stages should only have one key but found $group, $match.'
```

File: spec/AggregateRouter.spec.js [hunk 9]
```diff
@@ -37,55 +34,53 @@ describe('AggregateRouter', () => {
...
     ];
-    try {
-      AggregateRouter.getPipeline(body);
-    } catch (e) {
-      expect(e.message).toBe('Pipeline stages should only have one key found group, match');
-    }
+    expect(() => AggregateRouter.getPipeline(body)).toThrow(
+      new Parse.Error(
+        Parse.Error.INVALID_QUERY,
+        'Pipeline stages should only have one key but found $group, $match'
```
Comment (mtrezza) [inline spec/AggregateRouter.spec.js]:
```suggestion
        'Pipeline stages should only have one key but found $group, $match.'
```

**Review by mtrezza — CHANGES_REQUESTED** [review]
Punctuation added to error messages.

File: src/Routers/AggregateRouter.js [hunk 10]
```diff
@@ -83,22 +82,30 @@ export class AggregateRouter extends ClassesRouter {
...
   }
 
   static transformStage(stageName, stage) {
-    if (stageName === 'group') {
+    const skipKeys = ['distinct', 'where'];
+    if (skipKeys.includes(stageName)) {
+      return;
+    }
+    if (stageName[0] !== '$') {
+      throw new Parse.Error(Parse.Error.INVALID_QUERY, `Invalid aggregate stage: ${stageName}.`);
```
Comment (mtrezza) [inline src/Routers/AggregateRouter.js]:
```suggestion
      throw new Parse.Error(Parse.Error.INVALID_QUERY, `Invalid aggregate stage '${stageName}'.`);
```

**Review by mtrezza — DISMISSED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-alpha.19](https://github.com/parse-community/parse-server/releases/tag/6.0.0-alpha.19)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/6.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0](https://github.com/parse-community/parse-server/releases/tag/6.0.0)

**Pull request merged**